### PR TITLE
Upgrade to Preact 10

### PR DESCRIPTION
**Pull request by tjsilver**
## Why are you doing this?
We are keen to upgrade to Preact 10 so that we can implement the Source Design System components in our codebase.

Preact was upgraded to version 10 in [this PR](https://github.com/guardian/support-frontend/pull/2208/files) but it had to be [reverted](https://github.com/guardian/support-frontend/pull/2244) as it broke the Paypal button functionality on Contributions landing page. 

In this PR:

1. Upgraded to Preact 10
2. Fixed issue with Paypal button
3. Fixed new issue with Amazon Pay button

[**Trello card**](https://trello.com/c/yELvkq7V/1670-fix-bugs-with-preact-upgrade)

### CODE REVIEW

**Comment by jorgeazevedo** [comment]
Just pushed a fix for the selenium tests! The tests were failing locally

**Comment by jorgeazevedo** [comment]
I ran the selenium test suite against this branch (on CODE) using browser stack. I tested a number of browser and OS combinations, and everything seems to be working OK except IE 11. I'm currently investigating if it's a regression on our side, or a problem with identity, or what's happening.

**Comment by jorgeazevedo** [comment]
I've been able to reproduce the IE issue in production, so that issue is out of scope for this PR!

I've also tested (manually :-( ) subscriptions on a Pixel 2 and an iPhone. @tjsilver did recurring contributions. The only thing missing from my side is checking if the android app is still displaying the site in a webview to take contributions

File: support-frontend/assets/components/paypalExpressButton/PayPalExpressButton.jsx [hunk 1]
```diff
@@ -24,61 +26,48 @@ type PropTypes = {|
...
   setupRecurringPayPalPayment: SetupPayPalRequestType,
+  updatePayPalButtonReady: (boolean) => Action,
 |};
 
+const tokenToAuthorisation = (token: string): PayPalAuthorisation => ({
+  paymentMethod: PayPal,
+  token,
+});
 
-// Q. Why is this a class rather than a function?
```
Comment (tomrf1) [inline support-frontend/assets/components/paypalExpressButton/PayPalExpressButton.jsx]: For the record - we decided this was an unnecessary optimisation given the complexity involved in holding onto the same iframe (which is now more fragile with preact 10)

**Pull request merged**